### PR TITLE
Add snips built in intents 2016 12

### PR DESCRIPTION
**Pull request by bduvenhage**
This PR proposes to add the Snips.ai built in intents dataset. The first configuration added is for the intent labels only, but the dataset includes entity slots that may in future be added as alternate configurations.

### CODE REVIEW

**Comment by bduvenhage** [comment]
It is not clear how to automatically add the dummy data if the source data is a more complex json format. Should I manually take a fraction of the source data and include it as dummy data?

**Comment by bduvenhage** [comment]
Added a fraction of the real data as dummy data.

File: datasets/snips_built_in_intents/snips_built_in_intents.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+                        # Option 1 (Unshuffled): Yield samples here if un-shuffled samples ok.
+                        # yield num_examples, {"text": query_text, "label": label}
+
+                        num_examples += 1  # Explicitly keep track of the number of examples.
+
+        # Option 2 (Shuffled): Yield samples here if shuffled samples are required.
+        random.Random(12345).shuffle(
+            samples_txt_lbl
+        )  # Use a new instance of Random to fix shuffle pattern without affecting global rand.
```
Comment (lhoestq) [inline datasets/snips_built_in_intents/snips_built_in_intents.py:139]: could yo not shuffle the data ?

If the user wants to shuffle they can do `dataset.shuffle()`

**Review by lhoestq — COMMENTED** [review]
Nice thanks ! good job with the dummy data as well

I just left one comment about shuffling

File: datasets/snips_built_in_intents/snips_built_in_intents.py (hunk 1 above)
Comment (bduvenhage) [reply datasets/snips_built_in_intents/snips_built_in_intents.py:139]: Hi @lhoestq . I was hoping to have this discussion :-)
Yes, I don't need to shuffle. There is an 'option 1' a little higher up in the code that would then be the way to go. I'll uncomment that option and ping you again.

Comment (bduvenhage) [reply datasets/snips_built_in_intents/snips_built_in_intents.py]: @lhoestq You can have a look. If you are happy with the current version then we can remove all of the commented code.

**Review by lhoestq — APPROVED** [review]
Looks all good !

Thanks for adding this one :)

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**